### PR TITLE
#11 features: configure public live chat

### LINKED ISSUE #61: CI/TEST are failing
On "pull_request" and "push" the CI/TEST are failing

### PR DESCRIPTION
**Pull request by bryan936M**
#### What does this PR do?
- add bi-directional and group chatting capability to the project
#### Description of Task to be completed?
- GIVEN the user is online
WHEN they open the chat tab
THEN they should be able to read all of the past chats
- GIVEN the user is online
WHEN they open the chat tab
THEN they should be able to post a public chat
- GIVEN the user
WHEN they post a chat
THEN it should be able to be echoed to all online users in real-time.
#### How should this be manually tested?
- Download this repo locally
- Open it and run 'npm run dev'
- go to 'http://localhost:3000/chat/chatRoom'
- open multiple tabs of chatRoom
- login(use a user saved in the db)
- then text publicly or join/create a private room
#### Any background context you want to provide?
- This uses the SOCKET.IO API for managing communication
- Shops on the website can own a private room easily
#### What are the relevant pivotal tracker/Trello stories?
Trello card: https://trello.com/c/cnpHM5zi
#### Questions:

### CODE REVIEW

**Review by Muhet — APPROVED** [review]

**Review by Muhet — APPROVED** [review]

**Review by ALVINdimpos — APPROVED** [review]

**Review by faid-terence — APPROVED** [review]
Good Work

**Review by NTElissa — APPROVED** [review]
good progress

File: src/controller/chatController.js [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+
+class ChatController {
+  static async createRoom(req, res) {
+    try {
+      const { name } = req.body;
+      console.log(name);
+      const room = await db.Room.create({
+        name,
+      });
+      return res.status(201).json(room);
```
Comment (chanceDira) [inline src/controller/chatController.js:12]: Can you apply the Jsend format in this file

File: src/services/chatsDb.services.js [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+      const roomId = room.id;
+      const data = { userId, roomId, message };
+      const chat = await dbHandler.fetchJSON(
+        `${dbHandler.DOMAIN}/post/chat`,
+        'POST',
+        data,
+      );
+      return chat.id;
+    } catch (error) {
+      console.error('Error saving chat instance:', error);
```
Comment (chanceDira) [inline src/services/chatsDb.services.js]: You can remove the log messages in this file

**Review by chanceDira — CHANGES_REQUESTED** [review]
Nice job @bryan936M, just one thing

**Review by chanceDira — APPROVED** [review]
Good job @bryan936M

**Pull request merged**